Sources/NetworkProtection/Diagnostics/NetworkProtectionError.swift

```diff
@@ -77,6 +77,113 @@ public enum NetworkProtectionError: LocalizedError {
     // Unhandled error
     case unhandledError(function: String, line: Int, error: Error)
 
+    public static let errorDomain = "com.duckduckgo.NetworkProtectionError.domain"
+
+    public var errorCode: Int {
```

Comment:
One of the issues of adding `CustomNSError` is we need codes.  I still really like having associated values to I prefer to add these codes manually.

I tried to make them easy to modify in case we add anything to a specific section (without having to move all codes).